test/automate/automate-utils.js

```diff
@@ -14,285 +14,292 @@
+  // headless mode for chrome and firefox is enabled by default
+  let headless = true;
+  if (typeof process.env.HEADLESS !== 'undefined') {
+    // eslint-disable-next-line eqeqeq
+    headless = process.env.HEADLESS == true;
+  }
+
+  let capabilities;
+
+  if (browser === 'chrome') {
+
+    // Set up Chrome options.
+    const chrome = require('selenium-webdriver/chrome');
+    let chromeOptions = (new chrome.Options())
+      .addArguments([
+        'log-level=3',
+        'mute-audio',
+        'no-sandbox'
+      ])
+      .setChromeBinaryPath(chromeBinaryPath);
+    if (headless) {
+      chromeOptions.headless();
+    }
 
-  // Set up Chrome options. We use both "start-maximized" along with
-  // "window-size" because some platforms prefer one or the other.
-  let options = new chrome.Options();
-  options.addArguments([
-    'start-maximized',
-    `window-size=${browserMaxWidth}x${browserMaxHeight}`,
-    'no-sandbox'
-  ]);
-  options.setChromeBinaryPath(chromeBinaryPath);
+    capabilities = chromeOptions.toCapabilities();
+
+  } else if (browser === 'firefox') {
+
+    // Set up Firefox options.
+    const firefox = require('selenium-webdriver/firefox');
+    let profile = new firefox.Profile();
+    profile.setPreference('media.volume_scale', '0.0');
+    let firefoxOptions = (new firefox.Options())
+      .setProfile(profile)
+      .setBinary(firefoxBinaryPath);
+    if (headless) {
+      firefoxOptions.headless();
+    }
+
+    capabilities = firefoxOptions.toCapabilities();
+
+  } else {
+    throw new Error(`Browser ${browser} is not supported`);
+  }
+
+  capabilities.set('acceptInsecureCerts', true);
 
   // Create the browser.
-  var driver = new webdriver.Builder()
-    .forBrowser('chrome')
-    .setChromeOptions(options)
+  let driver = new webdriver.Builder()
+    .withCapabilities(capabilities)
     .build();
 
   // The following functions are defined below instead of outside this function
   // so we do not have to pass the 'driver' as a function parameter. It makes
   // the functions easier to use in a 'then' chain.
 
+  // Grab a screenshot and write to disk.
+  // TODO: Do we want to grab screenshots for success?
+  const saveScreenshot = (name, type) => {
+
+    const fileName =
+      `test-result-${screenShotUniq}-${browser}-` +
+      `${name.replace(/[ :]/g, '_')}-${type}.png`;
+
+    screenShotUniq++;
+
+    return driver.takeScreenshot()
+      .then(data => {
+        if (data) {
+          var base64Data = data.replace(/^data:image\/png;base64,/, '');
+          return fs.writeFile(
+            path.join(screenshotDir, fileName),
+            base64Data,
+            'base64',
+            err => {if (err) { console.log(err); }}
+          );
+        }
+        return null;
+      });
+  };
+
   // Locates an element and then clicks it.
-  const clickElement = function(locator) {
-    return driver.wait(
+  const clickElement = locator => driver.wait(
```

Comment:
This is a code improvement and isn't related to this PR. In the future you should put code improvements in a separate PR. If we have to reverse this PR we would lose the code improvement with it. I'm not going to let it stop us from merging this PR, but please put in the effort in the future to keep PRs very focused.